index.js

```diff
@@ -78,6 +83,196 @@ function loader (version) {
+          const durabilityFixedPerMat = Math.floor(MAX_DURABILITY * 0.25)
+          const matsToFullyRepair = Math.ceil(DURABILITY_LOST / durabilityFixedPerMat)
+          if (itemTwo.count > matsToFullyRepair) { // takeall of itemTwo
+            fixedDurability = MAX_DURABILITY
+            xpLevelCost = matsToFullyRepair // 1 exp lvl per mat used
+            usedMats = matsToFullyRepair
+          } else {
+            fixedDurability = DURABILITY_LOST + (durabilityFixedPerMat * itemTwo.count)
+            xpLevelCost = itemTwo.count // 1 exp lvl per mat used
+            usedMats = itemTwo.count
+          }
+        }
+        return [xpLevelCost, fixedDurability, usedMats]
+      }
+
+      /**
+         *
+         * @param {Item} itemOne
+         * @param {Item} itemTwo
+         * @param {boolean} creative
+         * @param {boolean} rename
+         */
+      function combine (itemOne, itemTwo, creative, rename) {
+        const data = {}
+        try {
+          combinePossible(itemOne, itemTwo) // throws if not possible
+          let cost = baseCost(itemOne) + baseCost(itemTwo)
+          if (rename) cost += renameCost(itemOne)
+          if (itemOne.metadata !== 0) {
+            // cost +=
+            const [xpLevelCost, fixedDurability, usedMats] = repairCost(itemOne, itemTwo)
+            data.fixedDurability = fixedDurability
+            data.usedMats = usedMats
+            cost += xpLevelCost
+          }
+          if (itemTwo.displayName === itemOne.displayName || itemTwo.name === 'enchanted_book') {
+            const { xpLevelCost: enchantCost, finalEnchs } = combineEnchants(itemOne, itemTwo, creative)
+            data.finalEnchs = finalEnchs
+            if (enchantCost === 0 && !rename && itemOne.metadata === 0) throw new Error('No change')
+            cost += enchantCost
+          }
+          return { xpCost: cost, ...data }
+        } catch (err) {
+          return { xpCost: 0, invalid: true, err: err.message } // errors should be handled by returning 0
+        }
+      }
+
+      function getMultipliers (weight) {
+        const itemMultiplier = {
+          10: 1,
+          5: 2,
+          2: 4,
+          1: 8
+        }[weight]
+        return { itemMultiplier, bookMultiplier: Math.max(1, itemMultiplier / 2) }
+      }
+
+      /**
+         *
+         * @param {Item} itemOne left hand item
+         * @param {Item} itemTwo right hand item
+         * @returns {[xpLevelCost, finalEnchs]}
+         * xpLevelCost is enchant data that is strictly from combining enchants
+         * finalEnchs is the array of enchants on the final object
+         */
+      function combineEnchants (itemOne, itemTwo, creative) {
+        const rightIsBook = itemTwo.name === 'enchanted_book'
+        const bothAreBooks = itemOne.name === 'enchanted_book' && rightIsBook
+        const findEnchantBy = (val, by) => Object.entries(mcData.enchantments).map(x => x[1]).find(x => x[by] === val)
```

Comment:
You can use https://github.com/PrismarineJS/node-minecraft-data/blob/master/lib/loader.js#L35